packages/react-animation/src/useFlip/useFlip.ts

```diff
@@ -0,0 +1,26 @@
+import { unref, type Unreffable } from '@mediamonks/react-hooks';
+import gsap from 'gsap';
+import Flip from 'gsap/Flip';
+import { useEffect, useRef, type MutableRefObject } from 'react';
+
+gsap.registerPlugin(Flip);
+
+export function useFlip(
+  ref: Unreffable<HTMLElement | null>,
+  flipStateVariables: Flip.FromToVars = {},
+): MutableRefObject<Flip.FlipState | undefined> {
+  const flipStateRef = useRef<Flip.FlipState>();
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (globalThis.window !== undefined) {
```

Comment:
It might be useful (for others, or future self), to add some comments explaining how/why this works.

I assume that:
- the `if` executes before potentially new style vars are set, capturing the "current" state.
- the `useEffect` executes after applying and rendering the new style vars, and then transitioning to them from the previously captured state.